File: app/src/main/java/com/funckyhacker/fileexplorer/di/AppModule.java
package com.funckyhacker.fileexplorer.di;

import com.funckyhacker.fileexplorer.util.PageManger;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module(subcomponents = {
    MainComponent.class,
    SearchComponent.class
})
public class AppModule {

  @Singleton @Provides
  public PageManger providePageManager() {
    return new PageManger();
  }
}

File: app/src/main/java/com/funckyhacker/fileexplorer/di/AppModule.kt
package com.funckyhacker.fileexplorer.di

import com.funckyhacker.fileexplorer.util.PageManger
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module(subcomponents = [
    (MainComponent::class),
    (SearchComponent::class)
])
class AppModule {

    @Singleton
    @Provides
    fun providePageManager(): PageManger {
        return PageManger()
    }
}
